### PR TITLE
My code review suggestion.

### PR DESCRIPTION
**Pull request by holtcl**
Analysis of Program:
This program is designed to retrieve compiled data from Covid-19 cases from across the globe and present them to the user in an organized fashion. Overall the program functioned well and appeared to be coded appropriately.

Was the Project Available in Github on Time:
Yes

Is the program documented well enough for you to understand:
Yes, the project was very well documented throughout

Does the Program Compile:
Yes

Rational Behind Changes
Changed several var variables to val as they were not modified and remained constant. I changed one instance of a variable country_Name to countryName to better fit Kotlin naming conventions.  I added a title to the bar chart to help better introduce what the data was, and I added a background color to another text view because I felt it helped separate the listview better and would be more visually appealing.  

Technical Suggestions:
Ensure that you are declaring variable appropriately i.e.  if a variable is going to remain constant use val instead of var.
Are you running enough Testing? There was only one test that was relevant to the project.
Is the Cases by Country going to add an unnecessary screen if you already have Country Cases listed on the main page? Or are you going to move the Country Cases from the bottom of the home page to a separate page, which would make the bar chart bigger and easier to read?

Technical Concepts I Learned:
How to import a barchart and place LiveData within it
How to create a page layout using CardViews
How to customize a page using both design view and code view together

### CODE REVIEW

File: app/src/main/java/edu/uc/groupproject/covid19tracker/dto/Cases.kt [hunk 1]
```diff
@@ -2,7 +2,7 @@ package edu.uc.groupproject.covid19tracker.dto
 
 import com.google.gson.annotations.SerializedName
 
-data class Cases(@SerializedName("country_name") var country_name: ArrayList<String>,
+data class Cases(@SerializedName("country_name") var countryName: ArrayList<String>,
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/dto/Cases.kt:5]: Changed to better fit Kotlin naming conventions https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/edu/uc/groupproject/covid19tracker/service/GlobalDataProvider.kt [hunk 2]
```diff
@@ -32,10 +32,10 @@ class GlobalDataProvider {
...
                      */
-                    var worldStat = response.body()!!.string()
-                    var Jobject = JSONObject(worldStat)
-                    var o = Jobject[dataType]
-                    var s = o as String
+                    val worldStat = response.body()!!.string()
+                    val Jobject = JSONObject(worldStat)
+                    val o = Jobject[dataType]
+                    val s = o as String
                     data.add(s)
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/service/GlobalDataProvider.kt:39]: Changed as the variables remained constant so the variable would be better off as a val

File: app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/ItemAdapter.kt [hunk 3]
```diff
@@ -30,7 +30,7 @@ class ItemAdapter(var context: Context, var view_layout: Int, var arrayCaseList:
         var deaths: TextView = view.findViewById<TextView>(R.id.list_deaths)
 
         var items: Cases = arrayCaseList.get(position)
-        title.text = items.country_name[0]
+        title.text = items.countryName[0]
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/ItemAdapter.kt:33]: Changed to better fit Kotlin naming conventions https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainFragment.kt [hunk 4]
```diff
@@ -113,7 +113,7 @@ class MainFragment : Fragment() {
              */
             for(x in 0 until 5) {
                 try{
-                    xAxisLabels.add(caseData.country_name[x])
+                    xAxisLabels.add(caseData.countryName[x])
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainFragment.kt:116]: Changed to better fit Kotlin naming conventions https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/edu/uc/groupproject/covid19tracker/dto/Cases.kt [hunk 5]
```diff
@@ -2,7 +2,7 @@ package edu.uc.groupproject.covid19tracker.dto
 
 import com.google.gson.annotations.SerializedName
 
-data class Cases(@SerializedName("country_name") var country_name: ArrayList<String>,
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/dto/Cases.kt:5]: Changed to better fit Kotlin naming conventions https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/ItemAdapter.kt [hunk 6]
```diff
@@ -30,7 +30,7 @@ class ItemAdapter(var context: Context, var view_layout: Int, var arrayCaseList:
         var deaths: TextView = view.findViewById<TextView>(R.id.list_deaths)
 
         var items: Cases = arrayCaseList.get(position)
-        title.text = items.country_name[0]
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/ItemAdapter.kt:33]: Changed to better fit Kotlin naming conventions https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainFragment.kt [hunk 7]
```diff
@@ -151,10 +151,10 @@ class MainFragment : Fragment() {
 
         fun setCountryListViewData(caseData: Cases) {
             val casesListViewItems = ArrayList<Cases>()
-            for(x in 5 until caseData.country_name.size) {
+            for(x in 5 until caseData.countryName.size) {
                 casesListViewItems.add(Cases(cases = arrayListOf(caseData.cases[x]), deaths = arrayListOf(caseData.deaths[x]),
                     totalRecovered = arrayListOf(caseData.totalRecovered[x]), activeCases = arrayListOf(caseData.activeCases[x]),
-                    newCases = arrayListOf(caseData.newCases[x]), country_name = arrayListOf(caseData.country_name[x]), newDeaths = arrayListOf(caseData.newDeaths[x]),
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainFragment.kt:157]: Changed to better fit Kotlin naming conventions https://kotlinlang.org/docs/reference/coding-conventions.html

File: app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainViewModel.kt [hunk 8]
```diff
@@ -32,54 +32,54 @@ class MainViewModel : ViewModel() {
...
-            var gCases: ArrayList<String>? = gdp.getGlobalCovidData("total_cases")
-            var gActive: ArrayList<String>? = gdp.getGlobalCovidData("active_cases")
-            var gTotalDeath: ArrayList<String>? = gdp.getGlobalCovidData("total_deaths")
-            var gRecovered: ArrayList<String>? = gdp.getGlobalCovidData("total_recovered")
-            var gNewCases: ArrayList<String>? = gdp.getGlobalCovidData("new_cases")
-            var gNewDeaths: ArrayList<String>? = gdp.getGlobalCovidData("new_deaths")
-            var gSeriousCritical: ArrayList<String>? = gdp.getGlobalCovidData("serious_critical")
-            var gCasesPerMillion: ArrayList<String>? = gdp.getGlobalCovidData("total_cases_per_1m_population")
-            var gDeathPerMillion: ArrayList<String>? = gdp.getGlobalCovidData("deaths_per_1m_population")
-            var gStatistics: ArrayList<String>? = gdp.getGlobalCovidData("statistic_taken_at")
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainViewModel.kt:49]: Changed as the variables remained constant so the variable would be better off as a val

File: app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainViewModel.kt [hunk 9]
```diff
@@ -32,54 +32,54 @@ class MainViewModel : ViewModel() {
...
              */
-            var country: ArrayList<String>? = cbc.getCasesByCountryData("country_name")
-            var cases: ArrayList<String>? = cbc.getCasesByCountryData("cases")
-            var deaths: ArrayList<String>? = cbc.getCasesByCountryData("deaths")
-            var active: ArrayList<String>? = cbc.getCasesByCountryData("active_cases")
-            var totalRecovered: ArrayList<String>? = cbc.getCasesByCountryData("total_recovered")
-            var newDeaths: ArrayList<String>? = cbc.getCasesByCountryData("new_deaths")
-            var newCases: ArrayList<String>? = cbc.getCasesByCountryData("new_cases")
-            var seriousCritical: ArrayList<String>? = cbc.getCasesByCountryData("serious_critical")
-            var totalCasesPerMillionPopulation: ArrayList<String>? = cbc.getCasesByCountryData("total_cases_per_1m_population")
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainViewModel.kt:76]: Changed as the variables remained constant so the variable would be better off as a val

File: app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainViewModel.kt [hunk 10]
```diff
@@ -32,54 +32,54 @@ class MainViewModel : ViewModel() {
...
+            val newDeaths: ArrayList<String>? = cbc.getCasesByCountryData("new_deaths")
+            val newCases: ArrayList<String>? = cbc.getCasesByCountryData("new_cases")
+            val seriousCritical: ArrayList<String>? = cbc.getCasesByCountryData("serious_critical")
+            val totalCasesPerMillionPopulation: ArrayList<String>? = cbc.getCasesByCountryData("total_cases_per_1m_population")
             delay(2000)
 
             /**
              * Set fetched country data to DTO object
              */
-            var casesDto = Cases(country!!, cases!!, deaths!!, active!!, totalRecovered!!, newDeaths!!,
```
Comment (holtcl) [inline app/src/main/java/edu/uc/groupproject/covid19tracker/ui/main/MainViewModel.kt:82]: Changed as the variables remained constant so the variable would be better off as a val

File: app/src/main/res/layout/main_fragment.xml [hunk 11]
```diff
@@ -30,6 +30,15 @@
...
         android:layout_margin="5dp">
 
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_margin="10dp"
+            android:text="Cases of Covid-19 by Country"
+            android:textColor="#1DA1F2"
+            android:textAlignment="center"
+            android:textSize="16dp" />
```
Comment (holtcl) [inline app/src/main/res/layout/main_fragment.xml:40]: Added title to bar graph for easier comprehension.

File: app/src/main/res/layout/main_fragment.xml [hunk 12]
```diff
@@ -132,21 +141,21 @@
...
             android:orientation="vertical">
 
             <TextView
-                android:layout_width="wrap_content"
+                android:layout_width="402dp"
                 android:layout_height="wrap_content"
-                android:textColor="#000"
-                android:textSize="16dp"
-                android:layout_margin="10dp"
-                android:text="Country Cases" />
```
Comment (holtcl) [inline app/src/main/res/layout/main_fragment.xml:143]: Added color to the TextView to for better separation between ListView and TextView which is more aesthetically pleasing

Comment (sukhbirsekhon) [reply app/src/main/res/layout/main_fragment.xml:143]: I disagree with using custom layout width instead of using wrap text because it will cause problems in different sizes of mobile devices

File: app/src/main/res/layout/main_fragment.xml [hunk 13]
```diff
@@ -132,21 +141,21 @@
...
+                android:background="#5B000000"
+                android:text="Country Cases"
+                android:textColor="#0B0000"
+                android:textSize="16dp" />
```
Comment (sukhbirsekhon) [inline app/src/main/res/layout/main_fragment.xml:153]: @holtcl we are currently keeping the UI of an application to a minimal level. We will start making it look aesthetically clean in the future. Don't need this right now.

File: app/src/main/res/layout/main_fragment.xml [hunk 14]
```diff
@@ -132,21 +141,21 @@
...
 
             <ListView
+                android:id="@+id/country_list_view"
                 android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:id="@+id/country_list_view">
-            </ListView>
```
Comment (sukhbirsekhon) [inline app/src/main/res/layout/main_fragment.xml:149]: @holtcl I am wondering why did you remove id?

File: app/src/main/res/layout/main_fragment.xml (hunk 11 above)
Comment (sukhbirsekhon) [reply app/src/main/res/layout/main_fragment.xml:40]: @holtcl I really appreciate your UI changes, but we are keeping the UI at minimal level. This block of code will become clutter in our codebase. We might change the text view in future.

**Review by sukhbirsekhon — CHANGES_REQUESTED** [review]

File: app/src/main/java/edu/uc/groupproject/covid19tracker/dto/Cases.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/groupproject/covid19tracker/dto/Cases.kt:5]: Agreed... good suggestion.

File: app/src/main/java/edu/uc/groupproject/covid19tracker/service/GlobalDataProvider.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/groupproject/covid19tracker/service/GlobalDataProvider.kt:39]: That's generally a good recommendation in Kotlin.  Declare as val by default, unless you believe the variable will be reassigned later.

File: app/src/main/res/layout/main_fragment.xml (hunk 12 above)
Comment (discospiff) [reply app/src/main/res/layout/main_fragment.xml:143]: Agree on that... better to keep the width flexible.

File: app/src/main/res/layout/main_fragment.xml (hunk 11 above)
Comment (discospiff) [reply app/src/main/res/layout/main_fragment.xml:40]: @sukhbirsekhon Good point here.  The goal is to minimize clutter/distractions on the screen, so everything on the screen should have a very good, defined purpose.  Otherwise, remove it.

**Comment by discospiff** [comment]
The naming conventions and var to val changes are good suggestions.  It looks like the group has already discussed these changes, so I'll leave the merge decision to them.
